tangelo/linq/target/target_stim.py

```diff
@@ -0,0 +1,150 @@
+            raise NotImplementedError("desired_meas_result not yet implemented with stim ")
+
+        if self.n_shots or self._noise_model:
+            translated_circuit = translate_c(source_circuit, "stim",
+                 output_options={"noise_model": self._noise_model})
+            for qubit in range(source_circuit.width):
+                translated_circuit.append("M", [qubit])
+            isamples = translated_circuit.compile_sampler().sample(self.n_shots)
+            samples = [''.join([str(int(q))for q in isamples[i]]) for i in range(self.n_shots)]
+            frequencies = {k: v / self.n_shots for k, v in Counter(samples).items()}
+        else:
+            self._current_state = translate_tableau(source_circuit).state_vector()
+            frequencies = self._statevector_to_frequencies(self._current_state)
+
+        return (frequencies, np.array(self._current_state)) if return_statevector else (frequencies, None)
+
+    def _get_expectation_value_from_statevector(self, qubit_operator, state_prep_circuit, initial_statevector=None, desired_meas_result=None):
+        """
+        Calculates the expectation value of a qubit operator using a TableauSimulator.
+
+        Args:
+            qubit_operator (QubitOperator): The qubit operator for which the expectation value is calculated.
+            state_prep_circuit (Circuit): The stabilizer circuit used to prepare the quantum state.
+            initial_statevector (list/array) : Not currently implemented, will raise an error
+            desired_meas_result (str) : Not currently implemented, will raise an error
+
+        Returns:
+            float: The real-valued expectation value of the qubit operator.
+
+        """
+        n_qubits = state_prep_circuit.width
+
+        if initial_statevector is not None:
+            raise NotImplementedError("initial_statevector not yet implemented with stim ")
+        if desired_meas_result is not None:
+            raise NotImplementedError("desired_meas_result not yet implemented with stim ")
+
+        if self.n_shots:
+            return self._get_expectation_value_from_frequencies(qubit_operator, state_prep_circuit)
+
+        else:
+            s = translate_tableau(state_prep_circuit)
+            paulisum = 0
+            for term, coef in qubit_operator.terms.items():
+                if len(term) > n_qubits:  # Cannot have a qubit index beyond circuit size
+                    raise ValueError(f"Size of operator {qubit_operator} beyond circuit width ({n_qubits} qubits)")
+                elif not term:  # Empty term: no simulation needed
+                    paulisum += coef
+                    continue
+                paulisum += coef * s.peek_observable_expectation(self.stim.PauliString(pauli_of_to_string(term, n_qubits)))
+        return np.real(paulisum)
+
+    def _get_expectation_value_from_frequencies(self, qubit_operator, state_prep_circuit, initial_statevector=None, desired_meas_result=None):
```

Comment:
since the code needs them and they are taken as args, we need to document initial_statevector and desired_meas_results. Just copy paste what you have above in previous method.

we'll see how we handle them in the next PR